doc/ipythonconsole.rst

```diff
@@ -2,91 +2,145 @@
+
+Any :guilabel:`IPython Console`, whether :ref:`external<connecting-external-kernel>` or started by Spyder, supports:
+
+* Automatic code completion
+* Real-time function calltips
+* Full GUI integration with the enhanced Spyder :doc:`debugging`.
+* The :doc:`variableexplorer`, with GUI-based editors for many built-in and third-party Python objects.
+* Display of Matplotlib graphics in Spyder's :doc:`plots` pane, if the :guilabel:`Inline` backend is selected under :menuselection:`Preferences --> IPython console --> Graphics --> Graphics backend`, and inline in the console if :guilabel:`Mute inline plotting` is unchecked under the :guilabel:`Plots` pane's options menu.
+
+.. image:: images/console/console-completion.png
+   :alt: Spyder IPython Console, with a popup list of code completion guesses
+
+For information on the features, commands and capabilities built into IPython itself, see the `IPython documentation`_.
+
+.. _IPython documentation: https://ipython.readthedocs.io/en/stable/overview.html
 
-.. note::
-   If on Windows and connecting to a remote machine over ``ssh``, you'll need to install the ``paramiko`` python package first, *e.g.* with ``conda install paramiko`` if using Anaconda.
 
-To connect to an external kernel,
 
-#. Launch an IPython kernel on the local or remote host if one is not already running.
+================
+Special consoles
+================
 
-   * If using Spyder 3.3.0 or later, you'll need to launch the kernel with ``python -m spyder_kernels.console`` (after you've first installed ``spyder-kernels`` on the host with ``<conda/pip> install spyder-kernels``).
-   * If using a version of Spyder before 3.3.0, ``ipython kernel`` should work to launch the kernel, albeit without certain Spyder-specific features.
+Spyder also supports several types of specialized consoles.
+A `Cython console`_ allows you to use the Cython language to speed up your code and call C functions directly from Python.
+A `Sympy console`_ enables creating and displaying symbolic math expressions right inside Spyder.
+Finally, a `Pylab console`_ loads common Numpy and Matplotlib functions by default; while this is deprecated and not recommended for new code, it can be used for older scripts that need it.
+
+.. _Cython console: https://cython.org/#documentation
+.. _Sympy console: https://docs.sympy.org/latest/index.html
+.. _Pylab console: https://matplotlib.org/faq/usage_faq.html#matplotlib-pyplot-and-pylab-how-are-they-related
+
+.. image:: images/console/console-special.gif
+   :alt: Spyder showing opening a new special Console
+
+
+
+============
+Options menu
+============
+
+The options menu allows you to inspect your current environment variables and the contents of your system's ``PATH`` in Spyder :doc:`variableexplorer` browsers with :guilabel:`Show environment variables` and :guilabel:`Show sys.path contents` respectively, and you can have each kernel display how long it has been running with :guilabel:`Show elapsed time`.
```

Comment:
```suggestion
The options menu allows you to inspect your current environment variables and the contents of your system's ``PATH`` with :guilabel:`Show environment variables` and :guilabel:`Show sys.path contents` respectively.
And you can have each console display how long it has been running with :guilabel:`Show elapsed time`.
```

* Although it's technically correct that we use the Variable Explorer viewers to display env vars and `sys.path` contents, I think that's an implementation detail users don't need to be informed about.
* I moved the last part to a new sentence because, in my opinion, the original sentence is too long.
* I think the concept of elapsed time would be easier to grasp if we talk about consoles and not kernels (although technically it's that time corresponds to the kernel).

Reply:
```suggestion
The options menu allows you to inspect your current environment variables and the contents of your system's ``PATH`` with :guilabel:`Show environment variables` and :guilabel:`Show sys.path contents` respectively.
And you can have each console display how long it has been running with :guilabel:`Show elapsed time`.
```

> I think that's an implementation detail users don't need to be informed about.

Fair enough; we should restructure it a bit to make it simpler and avoid a runon sentence.

> I moved the last part to a new sentence because, in my opinion, the original sentence is too long.

I agree (thought about it myself), but does need a grammar fix since "And" cannot start a sentence.

> I think the concept of elapsed time would be easier to grasp if we talk about consoles and not kernels

I mean, this isn't really correct since restarting a kernel within the same console will restart the time, but if you really feel this is an overly specific detail we can leave it out (since we do mention kernels some, but don't explicitly explain the distinction between the console/frontend and the kernel/backend like we used to).

```suggestion
The options menu allows you to inspect your current environment variables (:guilabel:`Show environment variables`), and the contents of your system's ``PATH`` (:guilabel:`Show sys.path contents`).
In addition, you can have each console display how long it has been running with :guilabel:`Show elapsed time`.
```